File: frontend/src/api/course.ts
import http from './index'

export interface Course {
  id?: number
  courseNo: string
  name: string
  credit: number
  hours: number
  type: string
  major?: string
  description?: string
  createdAt?: string
  updatedAt?: string
}

export function listCourses(params: {
  keyword?: string; type?: string; major?: string; page?: number; size?: number
}) {
  return http.get('/courses', { params }).then(r => r.data)
}

export function getCourse(id: number) {
  return http.get(`/courses/${id}`).then(r => r.data)
}

export function createCourse(data: Course) {
  return http.post('/courses', data).then(r => r.data)
}

export function updateCourse(id: number, data: Course) {
  return http.put(`/courses/${id}`, data).then(r => r.data)
}

export function deleteCourse(id: number) {
  return http.delete(`/courses/${id}`).then(r => r.data)
}

File: backend/src/main/java/com/sims/entity/Course.java
package com.sims.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("course")
public class Course {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String courseNo;
    private String name;
    private BigDecimal credit;
    private Integer hours;
    private String type;
    private String major;
    private String description;
    private Integer capacity;
    private Long teacherId;
    private LocalDate startDate;
    private LocalDate endDate;
    private String semester;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    @TableField(exist = false)
    private String teacherName;

    @TableField(exist = false)
    private Integer enrolledCount;
}
